From c6a5b75081382dafb57249a533d2feee68ba633f Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Mon, 1 Oct 2012 17:06:55 +0200
Subject: [PATCH 1/7] stream: Return error in case a client peeks to early

If there is no silence memblock and no data, pa_memblockq_peek can
return NULL. In this case, do not crash on an assertion in
pa_memblock_acquire, but instead return a proper error to the client.

BugLink: http://bugs.launchpad.net/bugs/1058200
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---
 src/pulse/stream.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/pulse/stream.c b/src/pulse/stream.c
index 8e35c29..a1bad27 100644
--- a/src/pulse/stream.c
+++ b/src/pulse/stream.c
@@ -1617,6 +1617,8 @@ int pa_stream_peek(pa_stream *s, const void **data, size_t *length) {
             return 0;
         }
 
+        PA_CHECK_VALIDITY(s->context, s->peek_memchunk.memblock != NULL, PA_ERR_NODATA);
+
         s->peek_data = pa_memblock_acquire(s->peek_memchunk.memblock);
     }
 
-- 
1.9.1

